src/utils/wrappers.py

```diff
@@ -0,0 +1,343 @@
+"""
+Copyright 2020 Google LLC.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import ee
+
+
+class CollectionClass:
+    """Base Imagery class for ImageCollection."""
+
+    def __init__(self, link, bands=None):
+        """Initialize the Collection class.
+
+        Parameters
+        ----------
+        link : str
+            ImageCollection link
+        bands : list[str], optional
+            Name of bands to select. The default is None to take all bands.
+
+        """
+        self.bands = bands
+        self.imagery = ee.ImageCollection(link)
+        if self.bands:
+            self.imagery = self.imagery.select(self.bands)
+
+    def get_clipped_bands(self, geometry):
+        """Get clipped images after a filter bound on geometry.
+
+        Parameters
+        ----------
+        geometry : ee.Geometry
+            Geometry to clip to
+
+        Returns
+        -------
+        ee.ImageCollection
+            cliped images filterbounded on geometry
+
+        """
+        images = self.imagery.filterBounds(geometry)
+        return images.map(lambda x: x.clip(geometry))
+
+    def get_clipped_range_bands(self, geometry, start_date, end_date):
+        """Get clipped images from a specific date range.
+
+        Parameters
+        ----------
+        geometry : ee.Geometry
+            DESCRIPTION.
+        start_date : String/EE.Date
+            DESCRIPTION.
+        end_date : String/EE.Date
+            DESCRIPTION.
+
+        Returns
+        -------
+        TYPE
+            DESCRIPTION.
+
+        """
+        images = self.imagery.filterBounds(geometry)
+
+        def clip_func(image):
+            return image.clip(geometry)
+        return images.filterDate(start_date, end_date).map(clip_func)
+
+
+class RoadImagery:
+    """Road Imagery Class: Wrapper for ee.Image."""
+
+    def __init__(self, link, bands):
```

Comment:
Did you mean to make this default? bands=None
If no, could you update the bands description below to reflect that this is required and does not have a default?